ee/modules/030-cloud-provider-vcd/crds/external/vcdaffinityrules.yaml

```diff
@@ -0,0 +1,112 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.18.0
+  labels:
+    heritage: deckhouse
+    module: cloud-provider-vcd
+  name: vcdaffinityrules.deckhouse.io
+spec:
+  group: deckhouse.io
+  names:
+    kind: VCDAffinityRule
+    listKind: VCDAffinityRuleList
+    plural: vcdaffinityrules
+    shortNames:
+    - vcdar
+    singular: vcdaffinityrule
+  scope: Cluster
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .status.nodeCount
+      name: NodeCount
+      type: integer
+    - jsonPath: .status.message
+      name: Message
+      type: string
+    name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: VCDAffinityRule is the Schema for the vcdaffinityrules API.
+        properties:
+          apiVersion:
+            description: |-
+              APIVersion defines the versioned schema of this representation of an object.
+              Servers should convert recognized schemas to the latest internal value, and
+              may reject unrecognized values.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+            type: string
+          kind:
+            description: |-
+              Kind is a string value representing the REST resource this object represents.
+              Servers may infer this from the endpoint the client submits requests to.
+              Cannot be updated.
+              In CamelCase.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: VCDAffinityRuleSpec defines the desired state of VCDAffinityRule.
+            properties:
+              nodeLabelSelector:
+                additionalProperties:
+                  type: string
+                description: |-
+                  NodeLabelSelector is a selector for the nodes that this rule applies to.
+                  Empty selector means that the rule applies to all nodes.
+                type: object
```

Comment:
nodeSelector is a key: value pairs, for nodeLabelSelector I expect it to be matchLabel / matchExpression selector like this

```
selector:
  matchLabels:
    component: redis
  matchExpressions:
    - { key: tier, operator: In, values: [cache] }
    - { key: environment, operator: NotIn, values: [dev] }
```